Products/ZenModel/actions.py

```diff
@@ -394,76 +395,104 @@ def _encodeBody(self, body):
             plain_body = MIMEText(body.decode('ascii', 'ignore'))
         return plain_body
 
+    def _targetsByTz(self, dmd, targets):
+        """
+        Take timezone from user property to convert a event time in
+        notification and also group targets emails by those timezones.
+        """
+        targets_by_timezones = {}
+        for user in dmd.ZenUsers.getAllUserSettings():
+            if user.email in targets:
+                if targets_by_timezones.get(user.timezone):
+                    targets_by_timezones[user.timezone].append(user.email)
+                else:
+                    targets_by_timezones[user.timezone] = [user.email]
+        return targets_by_timezones
+
     def executeBatch(self, notification, signal, targets):
         log.debug("Executing %s action for targets: %s", self.name, targets)
         self.setupAction(notification.dmd)
+        targets_by_timezones = self._targetsByTz(notification.dmd, targets)
+        original_lst = signal.event.last_seen_time
+        original_fst = signal.event.first_seen_time
+        for target_timezone in targets_by_timezones:
+            # Convert timestamp to user timezone
+            signal.event.last_seen_time = long(
+                isoToTimestamp(convertTimestampToTimeZone(
+                    (original_lst / 1000),
+                    target_timezone, '%Y-%m-%d %H:%M:%S'))) * 1000
+            signal.event.first_seen_time = long(
+                isoToTimestamp(convertTimestampToTimeZone(
+                    (original_fst / 1000),
+                    target_timezone, '%Y-%m-%d %H:%M:%S'))) * 1000
```

Comment:
I find this conversion code difficult to read and understand.

Are `original_fst` and `original_lst` just integer values that represent milliseconds?

Perhaps a lambda or helper function would help here, .e.g.
```
def adjustToTimezone(millis, timezone):
    return long(isoToTimestamp(convertTimestampToTimezone(
        millis / 1000, timezone, "%Y-%m-%d %H:%M:%S"
    ))) * 1000
```
and then later
```
        signal.event.first_seen_time = adjustToTimezone(
            signal.event.first_seen_time, target_timezone
        )
        signal.event.last_seen_time = adjustToTimezone(
            signal.event.last_seen_time, target_timezone
        )
```